### PR TITLE
35/fill in the sky with more stars

### PR DESCRIPTION
**Pull request by NunonuN**

### CODE REVIEW

File: src/components/canvas/Stars.js [hunk 1]
```diff
@@ -66,6 +66,186 @@ const generateStars = lampPosts => {
       opacity: 0.9 - numLampPosts * 0.1,
       r: 1.5,
     },
+    {
+      x: 1500,
+      y: 300,
+      opacity: 1 - numLampPosts * 0.1,
```
Comment (juancarlosfarah) [inline src/components/canvas/Stars.js]: @NunonuN, maybe it's a good time to factor out this `0.1` to either `settings.js` under a comment `// stars`, e.g.:

```
// stars
export const STAR_OPACITY_REDUCTION_FACTOR = 0.1;
```

Or at least a constant inside the same file.

Comment (juancarlosfarah) [reply src/components/canvas/Stars.js]: I would then have a helper function inside the file:

```
const calculateOpacity = numLampPosts => 1 - numLampPosts * STAR_OPACITY_REDUCTION_FACTOR;
```

And then you put `calculateOpacity(numLampPosts)` on the right-hand side of all those `opacity` keys.

**Comment by juancarlosfarah** [comment]
@NunonuN, as a sidenote, be careful about sending the PRs to `develop`. I saw that this one was going to `master`, so I edited it.

File: src/components/canvas/Stars.js (hunk 1 above)
Comment (NunonuN) [reply src/components/canvas/Stars.js]: This is a good idea. I will implement it.

**Comment by NunonuN** [comment]
Regarding the PR sent to `master`, that happended in Nairobi, by mistake, during our GO-GA meeting. Remember I told you about it? Don't worry, I'll be careful in future times. :wink:

File: src/components/canvas/Stars.js (hunk 1 above)
Comment (NunonuN) [reply src/components/canvas/Stars.js]: Done.

File: src/config/settings.js [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
-// area around a post that cannot include other posts
-export const BUFFER_WIDTH = 50;
-export const MAX_LAMP_POSTS = 7;
+// stars
+export const STAR_OPACITY_REDUCTION_FACTOR = 0.1;
 
+// miscellany
 export const DEFAULT_LANG = 'en';
-export const DEFAULT_MODE = 'student';
+export const DEFAULT_MODE = 'student'; // teacher
```
Comment (juancarlosfarah) [inline src/config/settings.js]: Remove the `// teacher` comment.

File: src/components/canvas/Stars.js [hunk 3]
```diff
@@ -1,251 +1,250 @@
-import React from 'react';
+import React from 'react'; // teacher
```
Comment (juancarlosfarah) [inline src/components/canvas/Stars.js]: Remove the `// teacher` comment.

File: src/components/canvas/Stars.js [hunk 4]
```diff
@@ -1,251 +1,250 @@
...
-      r: 1,
-    },
-    {
-      x: 300,
-      y: 150,
-      opacity: 0.7 - numLampPosts * 0.1,
-      r: 1,
+      x: 51,
+      y: 151,
+      opacity: 1 - calculateOpacity(numLampPosts),
```
Comment (juancarlosfarah) [inline src/components/canvas/Stars.js]: You could potentially factor out the `1 - ` to the `calculateOpacity` function as it always a part of the calculation.

**Pull request merged**